### ISSUE TITLE
Compile error when select Rule Override Tile in Tile Palette

### ISSUE THREAD

**Issue by tiomke** [Unity-Technologies/2d-techdemos#8]
Error ocurred when i was using 2018.2 branch with 2018.2.16f1.
Need help, Thanks!

```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.EditorUtility.InitInstantiatedPreviewRecursive (UnityEngine.GameObject go) (at C:/buildslave/unity/build/Editor/Mono/EditorUtility.cs:319)
UnityEditor.GridPaintPaletteWindow.ResetPreviewInstance () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:407)
UnityEditor.GridPaintPaletteClipboard.ResetPreviewInstance () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:276)
UnityEditor.GridPaintPaletteClipboard.OnAfterPaletteSelectionChanged () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:237)
UnityEditor.GridPaintPaletteWindow.set_palette (UnityEngine.GameObject value) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:267)
UnityEditor.GridPaintPaletteWindow.SelectPalette (Int32 i, System.Object o) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:840)
UnityEditor.FlexibleMenu.SelectItem (Int32 index) (at C:/buildslave/unity/build/Editor/Mono/GUI/FlexibleMenu/FlexibleMenu.cs:188)
UnityEditor.FlexibleMenu.OnGUI (Rect rect) (at C:/buildslave/unity/build/Editor/Mono/GUI/FlexibleMenu/FlexibleMenu.cs:129)
UnityEditor.PopupWindow.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/GUI/PopupWindow.cs:92)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:232)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MethodBase.cs:115)
UnityEditor.HostView.Invoke (System.String methodName, System.Object obj) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView.Invoke (System.String methodName) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView.OldOnGUI () (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:104)
UnityEngine.Experimental.UIElements.IMGUIContainer.DoOnGUI (UnityEngine.Event evt, Boolean isComputingLayout) (at C:/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:216)
```
```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.GridPaintPaletteClipboard.ScreenToGrid (Vector2 screenPosition) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:903)
UnityEditor.PaintableGrid.UpdateMouseGridPosition () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/PaintableGrid.cs:106)
UnityEditor.GridPaintPaletteClipboard.OnGUI () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:355)
UnityEditor.GridPaintPaletteWindow.OnClipboardGUI (Rect position) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:894)
UnityEditor.GridPaintPaletteWindow.OnGUI () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:356)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:232)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MethodBase.cs:115)
UnityEditor.HostView.Invoke (System.String methodName, System.Object obj) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView.Invoke (System.String methodName) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView.InvokeOnGUI (Rect onGUIPosition) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:255)

```
```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.EditorUtility.InitInstantiatedPreviewRecursive (UnityEngine.GameObject go) (at C:/buildslave/unity/build/Editor/Mono/EditorUtility.cs:319)
UnityEditor.GridPaintPaletteWindow.ResetPreviewInstance () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:407)
UnityEditor.GridPaintPaletteClipboard.ResetPreviewInstance () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:276)
UnityEditor.GridPaintPaletteClipboard.Render () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:581)
UnityEditor.GridPaintPaletteClipboard.OnGUI () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteClipboard.cs:379)
UnityEditor.GridPaintPaletteWindow.OnClipboardGUI (Rect position) (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:894)
UnityEditor.GridPaintPaletteWindow.OnGUI () (at C:/buildslave/unity/build/Modules/TilemapEditor/Editor/Managed/Grid/GridPaintPaletteWindow.cs:356)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:232)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MethodBase.cs:115)
UnityEditor.HostView.Invoke (System.String methodName, System.Object obj) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:295)
UnityEditor.HostView.Invoke (System.String methodName) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:288)
UnityEditor.HostView.InvokeOnGUI (Rect onGUIPosition) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:255)
```

**Comment by ChuanXin-Unity** [comment]
Hi, this has been fixed on the 2018.2 branch. The Rule Override Tiles palette was regenerated from 2018.2.

**Comment by tiomke** [comment]
@ChuanXin-Unity Perfect！It works now～ Thank you very much！
